classes/check/collectorcheck.php

```diff
@@ -0,0 +1,88 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Collector failure tests for Check API
+ *
+ * @package tool_cloudmetrics
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @copyright Catalyst IT
+ */
+
+namespace tool_cloudmetrics\check;
+
+use core\check\check;
+use core\check\result;
+use tool_cloudmetrics\collector\manager;
+
+class collectorcheck extends check {
+
+    /**
+     * Constructor
+     */
+    public function __construct() {
+        $this->id = 'checkcollectorcheck';
+        $this->name = get_string('checkcollectorcheck', 'tool_cloudmetrics');
+    }
+
+    /**
+     * A link to a place to action this
+     *
+     * @return action_link|null
+     */
+    public function get_action_link(): ?\action_link {
+        return new \action_link(
+            new \moodle_url('/admin/category.php?category=tool_cloudmetrics_reports'),
+            get_string('managelink', 'tool_cloudmetrics'));
+    }
+
+    /**
+     * Return result
+     * @return result
+     */
+    public function get_result() : result {
+
+        $failures = false;
+        $messages = [];
+
+        $tests = get_config('tool_cloudmetrics');
+        if ($tests) {
+            foreach ($tests as $key => $result) {
+                // Filter collector_status responses, ignore anything else.
+                if (strpos($key, manager::STATUS_PREFIX) === 0) {
+                    $name = substr($key, strlen(manager::STATUS_PREFIX));
+                    if ($result === 'pass') {
+                        $messages[] = get_string('collector_passed', 'tool_cloudmetrics', $name);
+                        continue;
+                    } else {
+                        $failures = true;
+                        $summarytemplate = get_string('collector_failed', 'tool_cloudmetrics', $name);
+                        // Fill in the time it first failed.
+                        $messages[] = sprintf($summarytemplate, userdate((int) $result));
```

Comment:
You don't need the sprintf,this can be one line as get_string can accept params directly eg

get_string('collector_failed', 'tool_cloudmetrics', $name, ['foo' => 'bar']);

'Metric \'{$a->foo}\' failed %s'

https://docs.moodle.org/dev/String_API#Run-time_parameters